hazelcast-sql-core/src/test/java/com/hazelcast/sql/SqlOrderByTest.java

```diff
@@ -0,0 +1,526 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.impl.SqlTestSupport;
+import com.hazelcast.test.HazelcastSerialParametersRunnerFactory;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static com.hazelcast.sql.SqlBasicTest.AbstractPojo;
+import static com.hazelcast.sql.SqlBasicTest.AbstractPojoKey;
+import static com.hazelcast.sql.SqlBasicTest.DataSerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.DataSerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.IdentifiedDataSerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.IdentifiedDataSerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.PortablePojo;
+import static com.hazelcast.sql.SqlBasicTest.PortablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.SerializablePojo;
+import static com.hazelcast.sql.SqlBasicTest.SerializablePojoKey;
+import static com.hazelcast.sql.SqlBasicTest.SerializationMode;
+import static com.hazelcast.sql.SqlBasicTest.portableFieldName;
+import static com.hazelcast.sql.SqlBasicTest.serializationConfig;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Test that covers basic column read operations through SQL.
+ */
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:RedundantModifier")
+public class SqlOrderByTest extends SqlTestSupport {
```

Comment:
I am not sure that this test covers sufficient number of cases. E.g.:
1. All supported types (e.g. these are no tests for `float`)
2. Handling of null values
3. Handling of the same values (e.g. 1, 2, 2)
4. Testing index lookups and ORDER BY together
5. Single member (no merge) and multiple members (with merge)

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18013/commits/f24b6ea9b5900c791ea30d8e430e0da71f9d4ff6